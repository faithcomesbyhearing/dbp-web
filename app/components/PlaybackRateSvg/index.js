/**
 *
 * PlaybackRateSvg
 *
 */

import React from 'react';
import PropTypes from 'prop-types';
import SvgWrapper from '../SvgWrapper';

function PlaybackRateSvg({ playbackRate }) {
	// If speed came from cookie it is stored as a string since there is not a parse float
	if (playbackRate === 0.75) {
		return <SvgWrapper className={'icon'} fill="#fff" svgid="playback_0.75x" />;
	} else if (playbackRate === 1) {
		return <SvgWrapper className={'icon'} fill="#fff" svgid="playback_1x" />;
	} else if (playbackRate === 1.25) {
		return <SvgWrapper className={'icon'} fill="#fff" svgid="playback_1.25x" />;
	} else if (playbackRate === 1.5) {
		return <SvgWrapper className={'icon'} fill="#fff" svgid="playback_1.5x" />;
	}
	return <SvgWrapper className={'icon'} fill="#fff" svgid="playback_2x" />;
}

PlaybackRateSvg.propTypes = {
	playbackRate: PropTypes.number,
};

export default PlaybackRateSvg;
