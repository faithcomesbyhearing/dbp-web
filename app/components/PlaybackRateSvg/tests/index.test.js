import React from 'react';
import { render } from '@testing-library/react';

import PlaybackRateSvg from '..';

describe('<PlaybackRateSvg /> component tests', () => {
	it('Should match snapshot with default props', () => {
		const { asFragment } = render(<PlaybackRateSvg playbackRate={1} />);

		expect(asFragment()).toMatchSnapshot();
	});
	it('Should match snapshot with 0.75 speed', () => {
		const { asFragment } = render(<PlaybackRateSvg playbackRate={0.75} />);

		expect(asFragment()).toMatchSnapshot();
	});
	it('Should match snapshot with 1.25 speed', () => {
		const { asFragment } = render(<PlaybackRateSvg playbackRate={1.25} />);

		expect(asFragment()).toMatchSnapshot();
	});
	it('Should match snapshot with 1.5 speed', () => {
		const { asFragment } = render(<PlaybackRateSvg playbackRate={1.5} />);

		expect(asFragment()).toMatchSnapshot();
	});
	it('Should match snapshot with 2 speed', () => {
		const { asFragment } = render(<PlaybackRateSvg playbackRate={2} />);

		expect(asFragment()).toMatchSnapshot();
	});
	it('Should match snapshot with no playback speed', () => {
		const { asFragment } = render(<PlaybackRateSvg />);

		expect(asFragment()).toMatchSnapshot();
	});
});
