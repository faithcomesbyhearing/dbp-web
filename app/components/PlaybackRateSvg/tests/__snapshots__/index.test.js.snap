// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PlaybackRateSvg /> component tests Should match snapshot with 0.75 speed 1`] = `
<svg
  className="icon"
  fill="#fff"
>
  <use
    xlinkHref="/svglist.svg#playback_0.75x"
    xmlnsXlink="http://www.w3.org/1999/xlink"
  />
</svg>
`;

exports[`<PlaybackRateSvg /> component tests Should match snapshot with 1.5 speed 1`] = `
<svg
  className="icon"
  fill="#fff"
>
  <use
    xlinkHref="/svglist.svg#playback_1.5x"
    xmlnsXlink="http://www.w3.org/1999/xlink"
  />
</svg>
`;

exports[`<PlaybackRateSvg /> component tests Should match snapshot with 1.25 speed 1`] = `
<svg
  className="icon"
  fill="#fff"
>
  <use
    xlinkHref="/svglist.svg#playback_1.25x"
    xmlnsXlink="http://www.w3.org/1999/xlink"
  />
</svg>
`;

exports[`<PlaybackRateSvg /> component tests Should match snapshot with 2 speed 1`] = `
<svg
  className="icon"
  fill="#fff"
>
  <use
    xlinkHref="/svglist.svg#playback_2x"
    xmlnsXlink="http://www.w3.org/1999/xlink"
  />
</svg>
`;

exports[`<PlaybackRateSvg /> component tests Should match snapshot with default props 1`] = `
<svg
  className="icon"
  fill="#fff"
>
  <use
    xlinkHref="/svglist.svg#playback_1x"
    xmlnsXlink="http://www.w3.org/1999/xlink"
  />
</svg>
`;

exports[`<PlaybackRateSvg /> component tests Should match snapshot with no playback speed 1`] = `
<svg
  className="icon"
  fill="#fff"
>
  <use
    xlinkHref="/svglist.svg#playback_2x"
    xmlnsXlink="http://www.w3.org/1999/xlink"
  />
</svg>
`;
